### ISSUE TITLE
Postal code relation imported as place

### ISSUE THREAD

**Issue by AndreKR** [osm-search/Nominatim#1979]
I am trying to extend import-street.style to include post code relations for Germany. I am working with the 3.5 Docker container. To find out how to add post code relations, I was comparing the [Street style](https://github.com/osm-search/Nominatim/blob/3.5.x/settings/import-street.style) to the [Full style](https://github.com/osm-search/Nominatim/blob/3.5.x/settings/import-full.style).

Here is a random post code relation from Bremen, Germany, in XML format:
```xml
<relation id="1086076" version="13" timestamp="2014-08-24T21:51:54Z">
  <member type="way" ref="71040934" role="outer"/>
  <member type="way" ref="72484170" role="outer"/>
  <member type="way" ref="72888062" role="outer"/>
  <member type="way" ref="71040964" role="outer"/>
  <member type="way" ref="299921439" role="outer"/>
  <member type="way" ref="299921440" role="outer"/>
  <member type="way" ref="290478933" role="outer"/>
  <member type="way" ref="290478935" role="outer"/>
  <member type="way" ref="71040952" role="outer"/>
  <member type="way" ref="72888055" role="outer"/>
  <member type="way" ref="72888060" role="outer"/>
  <member type="way" ref="71040966" role="outer"/>
  <tag k="note" v="28209 Bremen"/>
  <tag k="type" v="boundary"/>
  <tag k="source" v="http://wiki.openstreetmap.org/wiki/Import/Catalogue/Postleitzahlen_Deutschland_2010"/>
  <tag k="boundary" v="postal_code"/>
  <tag k="postal_code" v="28209"/>
  <tag k="postal_code_level" v="8"/>
</relation>
```

This gets imported into my `place` table like this:

![image](https://user-images.githubusercontent.com/1188538/94137745-373c3280-fe67-11ea-828c-d3a29342a92d.png)

As far as I [understand](https://github.com/osm-search/Nominatim/blob/master/docs/develop/Import.md) the `class` field is supposed to come from the `k` of a `tag` and the `type` field from the `v` if this is matched in the style file. But there is no `tag` with key `"place"`, so what is happening here?

**Comment by AndreKR** [comment]
Also... I now added
```json
{
    "keys" : ["boundary"],
    "values" : {
        "administrative" : "main",
        "postal_code": "main"
    }
},
```
to my style file, which did import the post code relation shown above, but without the actual post code. So in addition to that I also added
```json
{
    "keys" : ["postal_code", "postcode", "addr:postcode",
              "tiger:zip_left", "tiger:zip_right"],
    "values" : {
        "" : "postcode,fallback"
    }
},
```
and suddenly everything that has a post code gets imported, even when there is no matching `main` value in the style file.

For example this entry:
```xml
<way id="186021144" version="4" timestamp="2019-12-23T15:17:29Z">
    <nd ref="1967211807"/>
    <nd ref="1967211847"/>
    <nd ref="1967211865"/>
    <nd ref="1967211863"/>
    <nd ref="1967211864"/>
    <nd ref="1967211908"/>
    <nd ref="1967211826"/>
    <nd ref="1967211872"/>
    <nd ref="1967211925"/>
    <nd ref="1967211918"/>
    <nd ref="1967211927"/>
    <nd ref="1967211928"/>
    <nd ref="1967211930"/>
    <nd ref="1967211867"/>
    <nd ref="1967211807"/>
    <tag k="name" v="Amtliche Materialprüfungsanstalt Bremen"/>
    <tag k="office" v="yes"/>
    <tag k="building" v="yes"/>
    <tag k="addr:city" v="Bremen"/>
    <tag k="addr:street" v="Paul-Feller-Straße"/>
    <tag k="addr:country" v="DE"/>
    <tag k="addr:postcode" v="28199"/>
    <tag k="addr:housenumber" v="1"/>
</way>
```
gets added to my `place` table like this:

![image](https://user-images.githubusercontent.com/1188538/94142676-f2b49500-fe6e-11ea-996d-1bcdf0a0fa2c.png)

As you can see, again the class is `"place"` without a `place` tag actually appearing in the entry.

At the moment I don't know how to import post code relations (`boundary: postal_code`) but not buildings that happen to have a post code.

**Comment by lonvia** [comment]
Your solution is completely correct.

I recently noticed the issue with the `boundary=postal_code` being imported as `place=postcode` as well and fixed it as part of #1963 (more specifically in 7fb62ea904fc2354b0b281028e12af1f6d212562). The problem is that the style specified that boundaries should only be imported when they have a name (property `main,with_name`).

Regarding the postcode tags: the `fallback` property states that if there is no other usable information on the OSM object, Nominatim should still add the object as `place=postcode`. When you use the street style, it suddenly causes all address points with `addr:postcode` to be included. Leaving out the `fallback` only adds the postcode information for objects that are included anyway (including the postal boundaries). That is exactly what you want.
